### DISCUSSION TITLE
eslint-config-next on ESlint v9 example

### DISCUSSION THREAD

**Discussion by satu-jovanhartono** [vercel/next.js#71806] [Help] [answered]
### Summary

ESlint version 9 is supported, starting on version 15
https://nextjs.org/blog/next-15#eslint-9-support 

any example of how the config file would look like?
i am trying to import the config on `eslint.config.mjs` filem but got this error when running `next lint`:
`Failed to patch ESLint because the calling module was not recognized.
If you are using a newer ESLint version that may be unsupported, please create a GitHub issue:
https://github.com/microsoft/rushstack/issues
 ELIFECYCLE  Command failed with exit code 1.`

### Additional information

```js
import nextlint from "eslint-config-next";

export default [
  {files: ["**/*.{js,mjs,cjs,ts,jsx,tsx}"]},
  nextlint,
];
```

**Comment by joepetrillo** [comment, marked as answer]
Use the `npx @eslint/migrate-config .eslintrc.json` command.

It produces ugly output, but seems to work - I assume we will eventually be able to do it as recommended in the [eslint docs](https://eslint.org/docs/latest/use/configure/configuration-files#using-a-shareable-configuration-package).

Heres the output I got (placed inside `eslint.config.mjs` file).

```js
import path from "node:path";
import { fileURLToPath } from "node:url";
import js from "@eslint/js";
import { FlatCompat } from "@eslint/eslintrc";

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);
const compat = new FlatCompat({
    baseDirectory: __dirname,
    recommendedConfig: js.configs.recommended,
    allConfig: js.configs.all
});
export default [...compat.extends("next")];
```
